### ISSUE TITLE
3rd-party: wrong paths to app resources

### ISSUE THREAD

**Issue by paulcarroty** [clearlinux/swupd-client#1443]
When app located in `/opt/..` trying to read his resources in `/usr/share/$APPNAME/` and can't find them 'cause they're in `/opt/..` too.

**Comment by otaviobp** [comment]
Thank you for reporting that.
At first, swupd 3rd-party was only supporting binaries and libraries from a 3rd-party repository to be used.
In swupd 3.24.5 XDG_DATA_DIRS and XDG_CONF_DIRS are filled with the correct path on /opt.
Can you check if this solves your problem? Besides updating swupd, you need to repair or reinstall the bundles that you created, so run `sudo swupd 3rd-party repair` after updating.

**Comment by paulcarroty** [comment]
Upgraded to the latest release & reinstalled all bundles to be sure, `XDG_CONFIG_DIRS` not included  `/opt`:

```
$ env | grep -i xdg
XDG_CONFIG_DIRS=/usr/share/xdg:/etc/xdg
XDG_MENU_PREFIX=gnome-
XDG_SESSION_DESKTOP=gnome-wayland
XDG_SESSION_TYPE=wayland
XDG_CURRENT_DESKTOP=GNOME
XDG_SESSION_CLASS=user
XDG_RUNTIME_DIR=/run/user/1000
XDG_DATA_DIRS=/home/paul/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/opt/3rd-party/bundles/clearfraction/usr/share/
```

And the problem still exists.

**Comment by otaviobp** [comment]
This is a bug then, I'll change the labels.

**Comment by otaviobp** [comment]
There are 2 considerations about this bug.

First, on binary wrappers we are adding XDG_CONFIG_DIRS with the incorrect name so this is a real bug and I'm going to fix that for next release.
Another point is that XDG variables are not being updated for the system, so system applications or applications for other 3rd-party repositories are not going to be able to access XGD files installed by 3rd-party. We don't have a plan to add that for now, but we can reconsider if there are use cases for that. Is this the solution you are expecting or the first case would solve your problem?

**Comment by otaviobp** [comment]
I just realized we actually have an issue to track my 2nd point: #1420

**Comment by paulcarroty** [comment]
>  Is this the solution you are expecting

Not sure if env variables can solve this issue. IRC `#clearlinux` people said the fix will look like a derivative of "fakeroot" or mounting namespace.

> We don't have a plan to add that for now, but we can reconsider if there are use cases for that.

A lot of use cases, especially if 3rd packages are libraries or frameworks, like gstreamer or ffmpeg. Global `XDG*` and `LD_LIBRARY_PATH` tuning is necessary to make them work with other applications.

#1420 tracks desktop entries & icons.

**Comment by bryteise** [comment]
> Global XDG* and LD_LIBRARY_PATH tuning is necessary to make them work with other applications.

@paulcarroty We definitely value feedback that the use cases of libraries and plugins are important for users and it is something that will impact our planning. Today users can enable this for themselves by configuring the environments the additional paths but it is more manual work.

Certain problems do require different solutions than using environment variables as you noted and probably getting a citing a specific 3rd-party application could improve our ability to test our solutions to this problem.

**Comment by otaviobp** [comment]
I also created another issues to track the export of resources on /usr/share from 3rd-party repositories (#1463) and one for exporting libraries (#1464).

**Comment by paulcarroty** [comment]
@bryteise [the application](https://gitlab.gnome.org/World/Shortwave), [rpm config](https://github.com/clearfraction/shortwave/blob/master/shortwave.spec),  and the resource [loading](https://gitlab.gnome.org/World/Shortwave/-/blob/master/src/main.rs#L82).

**Comment by otaviobp** [comment]
Variable is fixed on swupd 4.0. Closing this issue.
